jocktos/inc/os.h

```diff
@@ -52,29 +78,56 @@ typedef struct {
 void createTask(T_TaskControlBlock* tcb);
 
 /**
-* \brief Switch the currently running task
-*
-* Updates the Schedulers linked lists and task states upon context switch 
-*
-* \return
-*/
-void switchRunningTask(void);
+ * \brief Switch the currently running task
+ *
+ * Updates the Schedulers linked lists and task states upon context switch 
+ *
+ * \param head Pointer to destination for current running task.
+ */
+void switchRunningTask(volatile T_TaskControlBlock** head);
 
 /**
-* \brief Updates the task control blocks stackUsage
-* \return
-*/
+ * \brief Updates the task control blocks stackUsage
+ * 
+ * \param tcb Pointer to task control block to be monitored.
+ */
 static inline void monitorStackUsage(volatile T_TaskControlBlock** tcb) {
     (*tcb)->stackUsage = 100.0 * (1.0 - ((double)((*tcb)->u32TaskStackPointer \
-    - (*tcb)->u32TaskStackOverflow)) / (double)((*tcb)->u32StackSize_By * sizeof(uint32_t)));
+    - (*tcb)->u32TaskStackOverflow)) / (double)((*tcb)->u32StackSize_By * sizeof(uintptr_t)));
 }
+
 /**
-* \brief Enable scheduler and context switching ISR's
+* \brief pre defined OS task for idle.
 *
-* Sets the priorities and enables systick and pendSV handlers
-*
-* \return
+* Infinite while loop.
+* TODO: Figure out how to low power sleep without disabling ISR's

```

Comment:
```suggestion
WFI()
```
Simplest thing that we can do is just to include that into the scheduler. It will sleep until the next interrupt, and when no tasks are active. 

We could just try this for now since issues with memory retention and wakeup sources will be something we need to consider before diving deeper.

Some deeper sleep states will incur a small delay before becoming fully active again as clock sources re-stabilize, etc.